gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/CgmesExportService.java

```diff
@@ -239,19 +238,29 @@ private void addPTolerance(XMLStreamReader reader, XMLStreamWriter writerStream)
         }
     }
 
-    private Map<String, ByteArrayOutputStream> createOneTp(Network network, SweData sweData, String country) throws IOException, XMLStreamException {
-        try (ByteArrayOutputStream os = new ByteArrayOutputStream()) {
-            XMLStreamWriter writer = XmlUtil.initializeWriter(true, INDENT, os);
-            TopologyExport.write(network, writer, new CgmesExportContext(network));
-            return Map.of(buildCgmesFilename(sweData, country, "TP"), os);
-        }
+    private Map<String, ByteArrayOutputStream> createOneFile(SweData sweData, String country, String type) throws IOException {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        IOUtils.copy(getInputStreamFromData(sweData, country, type), os);
+        os.close();
```

Comment:
Why not use try-with-resources to ensure the closing of the BAOS even in case of exception during the copy?